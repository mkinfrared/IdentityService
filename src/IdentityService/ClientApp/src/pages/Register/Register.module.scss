@import "../../styles/common";

.Register {
  @extend %loginRegisterPage;

  .form {
    justify-content: flex-start;
  }
}

.Story {
}
