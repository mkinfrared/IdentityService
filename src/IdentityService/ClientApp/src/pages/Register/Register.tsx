import { memo } from "react";

import RegisterForm from "components/RegisterForm";
import { useReturnUrl } from "hooks";

import css from "./Register.module.scss";

const Register = () => {
  const returnUrl = useReturnUrl();

  return (
    <div className={css.Register}>
      <RegisterForm className={css.form} returnUrl={returnUrl} />
    </div>
  );
};

export { Register };

export default memo(Register);
