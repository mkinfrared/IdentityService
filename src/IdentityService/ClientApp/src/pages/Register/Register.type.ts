export type RegisterProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
