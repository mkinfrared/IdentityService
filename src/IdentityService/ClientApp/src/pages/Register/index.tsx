import ReactDOM from "react-dom";

import RegisterFC from "./Register";

import "@identity-service/ui/lib/styles.min.css";
import "index.scss";

ReactDOM.render(<RegisterFC />, document.getElementById("client-app"));
