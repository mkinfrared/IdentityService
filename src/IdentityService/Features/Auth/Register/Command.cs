using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Register
{
    public partial class Register
    {
        public class Command : IRequest<IdentityResult>
        {
            public Command(string username, string firstName, string lastName, string email,
                string phoneNumber, string password, string passwordConfirmation, string redirectUrl)
            {
                Username = username;
                FirstName = firstName;
                LastName = lastName;
                Email = email;
                PhoneNumber = phoneNumber;
                Password = password;
                PasswordConfirmation = passwordConfirmation;
                RedirectUrl = redirectUrl;
            }

            public string Username { get; }
            public string FirstName { get; }
            public string LastName { get; }
            public string Email { get; }
            public string PhoneNumber { get; }
            public string Password { get; }
            public string PasswordConfirmation { get; }
            public string RedirectUrl { get; }
        }
    }
}
