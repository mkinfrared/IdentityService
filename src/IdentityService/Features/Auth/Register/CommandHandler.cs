using System.Threading;
using System.Threading.Tasks;

using IdentityService.Entities;

using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Register
{
    public partial class Register
    {
        public class CommandHandler : IRequestHandler<Command, IdentityResult>
        {
            private readonly UserManager<User> _userManager;

            public CommandHandler(UserManager<User> userManager)
            {
                _userManager = userManager;
            }

            public async Task<IdentityResult> Handle(Command request,
                CancellationToken cancellationToken)
            {
                var user = new User
                {
                    UserName = request.Username,
                    FirstName = request.FirstName,
                    LastName = request.LastName,
                    Email = request.Email,
                    PhoneNumber = request.PhoneNumber
                };

                var result = await _userManager.CreateAsync(user, request.Password);

                // TODO add email confirmation send on register
                // Do i need to sign in user on register?
                return result;
            }
        }
    }
}
