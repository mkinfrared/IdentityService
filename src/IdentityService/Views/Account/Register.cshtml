@page
@model object

@{
    ViewBag.Title = "Login";
    ViewBag.Error = "Error message";
    Layout = "_Layout";
}

@section Head
{
    <link asp-href-include="~/register.*.css" rel="stylesheet">
}

@section PostBody
{
    <script type="application/javascript" asp-src-include="~/register.*.js"></script>
}

<div id="client-app">
    Loading...
</div>
