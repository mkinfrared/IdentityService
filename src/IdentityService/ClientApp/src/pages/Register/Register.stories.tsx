import { Meta, Story } from "@storybook/react";

import { Register } from "./Register";
import css from "./Register.module.scss";
import { RegisterProps } from "./Register.type";

export default {
  title: "PAGES/Register",
  component: Register,
  parameters: {
    componentSubtitle: "Subtitle goes here"
  }
} as Meta;

const Template: Story<RegisterProps> = () => (
  <div className={css.Story}>
    <Register />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description"
  }
};

export { Default };
