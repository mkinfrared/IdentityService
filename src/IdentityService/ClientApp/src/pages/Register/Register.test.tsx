import { render } from "@testing-library/react";

import { Register } from "./Register";

describe("<Register />", () => {
  const Component = <Register />;

  it("should be defined", () => {
    expect(Register).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
