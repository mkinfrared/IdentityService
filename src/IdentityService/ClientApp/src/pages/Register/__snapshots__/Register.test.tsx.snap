// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should match the snapshot 1`] = `
<div>
  <div
    class="Register"
  >
    <div
      class="Card-module_Card__2zjDp RegisterForm form"
      data-testid="Card"
    >
      <form>
        <h3
          class="Heading-module_Heading__iIC9P heading"
          data-testid="Heading"
        >
          Register
        </h3>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Username
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="username"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            First Name
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="firstName"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Last Name
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="lastName"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Email
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="email"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Phone Number
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="phoneNumber"
              type="text"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input TextField-module_hasSuffix__3ZA6e"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Password
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="password"
              type="password"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </div>
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <label
          class="TextField-module_TextField__3-bgB input TextField-module_hasSuffix__3ZA6e"
          data-testid="TextField"
        >
          <h5
            class="Heading-module_Heading__iIC9P TextField-module_label__SBMQi"
            data-testid="Heading"
          >
            Confirm Password
          </h5>
          <div
            class="TextField-module_input__3FXIK"
          >
            <input
              name="passwordConfirmation"
              type="password"
              value=""
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_prefix__2eBJg"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            />
            <div
              class="FakeButton-module_FakeButton__3fGxh TextField-module_suffix__2X_LC"
              data-testid="FakeButton"
              role="button"
              tabindex="0"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </div>
          </div>
          <p
            class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
            data-testid="Text"
          />
        </label>
        <input
          hidden=""
          name="redirectUrl"
          readonly=""
          type="text"
          value=""
        />
        <div
          class="buttonWrapper"
        >
          <button
            class="Button-module_Button__2YI99 Button-module_opaque__1DMAu"
            data-testid="Button"
            disabled=""
          >
            Register
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
